app/uk/gov/hmrc/cdsreimbursementclaimfrontend/controllers/Router.scala

```diff
@@ -129,17 +133,30 @@ trait JourneyTypeRoutes extends Product with Serializable {
       else claimRoutes.CheckYourAnswersAndSubmitController.checkAllAnswers()
     }
 
-  def nextPageForClaimantDetails(): Call =
-    claimRoutes.EnterDetailsRegisteredWithCdsController.enterDetailsRegisteredWithCds()
-
-  // TODO: There are two scenarios for claimant details next page:
-  //  If there are no contact details => the next page is EnterOrChangeMrnContactDetails.enterMrnContactDetails()
+  def nextPageForAddClaimantDetails(anwer: CheckClaimantDetailsAnswer): Call =
+    anwer match {
+      case YesClaimantDetailsAnswer =>
+        claimRoutes.EnterOrChangeContactDetailsController.enterMrnContactDetailsSubmit(journeyBindable)
+      case NoClaimantDetailsAnswer  =>
+        claimRoutes.EnterDetailsRegisteredWithCdsController.enterDetailsRegisteredWithCds()
+      //TODO put this back after removing the EnterDetailsRegisteredWithCdsController
+      //        featureSwitch.NorthernIreland.isEnabled() match {
+      //          case true  => claimRoutes.ClaimNorthernIrelandController.selectNorthernIrelandClaim(journeyBindable)
+      //          case false => claimRoutes.SelectBasisForClaimController.selectBasisForClaim(journeyBindable)
+      //        }
```

Comment:
Why remove EnterOrChangeMrnContactDetails?

Reply:
These lines have nothing to do with that. What am I missing?

Reply:
//TODO put this back after removing the EnterDetailsRegisteredWithCdsController